scripts/create_role_molecule.py

```diff
@@ -0,0 +1,124 @@
+import os
+import re
+from pathlib import Path
+from jinja2 import Environment, FileSystemLoader, TemplateError, TemplateNotFound
+
+
+PROJECT_DIR = Path(__file__).resolve().parent.parent
+SCRIPT_DIR = PROJECT_DIR / 'scripts'
+TEMPLATES_DIR = PROJECT_DIR / 'ci' / 'templates'
+ROLES_DIR = PROJECT_DIR / 'ci_framework' / 'roles'
+ZUUL_JOB_DIR = PROJECT_DIR / 'zuul.d'
+SOURCE_ROLES_DOCS_DIR = PROJECT_DIR / 'docs' / 'source' / 'roles'
+
+# These roles will use centos-9-crc-xl nodeset and it will be used to replace the content 
+# of the Jinja template.
+ROLES_NEED_CRC_XML = [
```

Comment:
How are gonna provide custom properties like timeouts?

Reply:
maybe getting a YAML file as a source of customization is a thing?

i.e:
```YAML
# custom-molecule.yml
operator_deploy:
  nodeset: crc_xxl
  timeout: 4800
rhol_crc:
  nodeset: crc_xxl
....
```

nested in the `ci` directory, and used in that script. And if a job (or specific parameters) aren't defined there, we'll fallback on the default things.

Reply:
Exactly, that was the idea we discussed, seems like easy to do and maintain